analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,742 @@
+Since we are plotting the proliferative markers, let's also look at our other custom marker gene sets. 
+
+```{r}
+# get columns with mean expression of custom gene sets 
+mean_exp_columns <- colnames(all_info_df)[which(endsWith(colnames(all_info_df), "_mean"))]
+# make sure order is defined by cell type order
+all_info_df <- all_info_df |> 
+  dplyr::arrange(final_lumped)
+
+single_annotation_heatmap(
+  all_info_df, 
+  exp_columns = mean_exp_columns, 
+  cell_type_column = "final_lumped", 
+  legend_title = "Mean expression"
+)
+```
+
+It looks like we do have a set of cells that are `tumor EWS-high` and also have high expression of proliferative markers. 
+Because of this, let's label any cells that are `tumor EWS-high` and have mean expression of proliferative markers > 0 as `tumor EWS-high proliferative`. 
+
+```{r}
+# set desired order for plotting final annotations later
+cell_type_order <- c(
+  "tumor EWS-high",
+  "tumor EWS-high proliferative",
+  "tumor EWS-low",
+  "muscle cell", 
+  "smooth muscle cell",
+  "fibroblast",
+  "endothelial cell",
+  "macrophage",
+  "mature T cell",
+  "Unknown",
+  "All remaining cell types"
+)
+
+all_info_df <- all_info_df |> 
+  dplyr::mutate(
+    final_annotation = dplyr::if_else(
+      final_annotation == "tumor EWS-high" & proliferative_mean > 0,
+      "tumor EWS-high proliferative",
+      final_annotation
+    ), 
+    final_lumped = final_annotation |>
+      forcats::fct_lump_n(10, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel(cell_type_order)
+  )
+
+# first get tally for how many libraries each cell type is present in 
+total_libraries <- all_info_df |> 
+  dplyr::select(final_annotation, library_id) |> 
+  unique() |> # get rid of duplicate cells
+  dplyr::group_by(final_annotation) |> 
+  dplyr::summarize(
+    total_libraries = length(library_id)) |> 
+  dplyr::arrange(desc(total_libraries))
+
+# print out both total cells and total libraries 
+all_info_df |> 
+  dplyr::count(final_annotation, name = "total_cells") |>
+  dplyr::left_join(total_libraries, by = "final_annotation") |> 
+  dplyr::arrange(desc(total_cells))
+```
+
+
+For plotting purposes, we will only show the top 10 cell types with muscle cell being the last cell type. 
+It looks like all other cell types are only present in 1-3 libraries. 
+For any cells where there's only 1 cell we may want to confirm that those cells express the expected targets, but that is outside the scope of this notebook. 
+
+## Validation of cell type assignments
+
+In the following section we'll show some plots to help validate that we are content with our cell type assignments. 
+These plots will look specifically at expression of a set of key marker genes that we expect to up in the assigned cell types. 
+These markers are found in `references/combined-markers.tsv` and include a smaller set of markers for each of the cell types we have identified. 
```

Comment:
Doesn't exist _yet_, but it will (also fixing the name)
```suggestion
These markers are found in [`references/combined-validation-markers.tsv`](https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/cell-type-ewings/references/combined-validation-markers.tsv) and include a smaller set of markers for each of the cell types we have identified. 
```